### PR TITLE
feat/#455 impl ErrorOutOfGasStaticMemoryExpansion error state

### LINKED ISSUE #455: Implement `ErrorOutOfGasStaticMemoryExpansion` error state
See https://github.com/privacy-scaling-explorations/zkevm-specs/blob/master/specs/error_state/ErrorOutOfGasStaticMemoryExpansion.md for more details

### PR DESCRIPTION
**Pull request by KimiWu123**
fixed #455

### CODE REVIEW

File: src/zkevm_specs/evm_circuit/execution/error_oog_static_memory_expansion.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    )
+
+    # Constrain opcode must be one of MLOAD, MSTORE or MSTORE8.
+    instruction.constrain_equal(is_mload + is_mstore + is_mstore8, FQ(1))
+
+    # pop `offset`
+    offset = instruction.word_to_fq(instruction.stack_pop(), N_BYTES_MEMORY_ADDRESS)
+
+    # get total gas cost
+    constant_gas = FQ(3)
```
Comment (hero78119) [inline src/zkevm_specs/evm_circuit/execution/error_oog_static_memory_expansion.py:22]: will it better to use `GAS_COST_FASTEST` in param.py to replace this constant_gas cost

**Review by hero78119 — APPROVED** [review]
LGTM! 

an nitpick on a constant usage

File: src/zkevm_specs/evm_circuit/execution/error_oog_static_memory_expansion.py (hunk 1 above)
Comment (KimiWu123) [reply src/zkevm_specs/evm_circuit/execution/error_oog_static_memory_expansion.py]: fixed in [a84ae0a](https://github.com/privacy-scaling-explorations/zkevm-specs/pull/474/commits/a84ae0a23eca01a64967e0ac59ef69429506899e)

**Review by rrtoledo — APPROVED** [review]
LGTM :)

**Pull request merged**